beacon_node/http_api/src/lib.rs

```diff
@@ -1857,15 +1904,29 @@ pub fn serve<T: BeaconChainTypes>(
         );
 
     // GET beacon/pool/attester_slashings
-    let get_beacon_pool_attester_slashings = beacon_pool_path
+    let get_beacon_pool_attester_slashings = beacon_pool_path_any
         .clone()
         .and(warp::path("attester_slashings"))
         .and(warp::path::end())
         .then(
-            |task_spawner: TaskSpawner<T::EthSpec>, chain: Arc<BeaconChain<T>>| {
-                task_spawner.blocking_json_task(Priority::P1, move || {
-                    let attestations = chain.op_pool.get_all_attester_slashings();
-                    Ok(api_types::GenericResponse::from(attestations))
+            |endpoint_version: EndpointVersion,
+             task_spawner: TaskSpawner<T::EthSpec>,
+             chain: Arc<BeaconChain<T>>| {
+                task_spawner.blocking_response_task(Priority::P1, move || {
+                    let slashings = chain.op_pool.get_all_attester_slashings();
+                    let slot = slashings
+                        .first()
+                        .map(|slashing| slashing.attestation_1().data().slot)
+                        .or_else(|| chain.slot_clock.now())
+                        .ok_or(warp_utils::reject::custom_server_error(
+                            "unable to read slot clock".to_string(),
+                        ))?;
+                    let fork_name = chain.spec.fork_name_at_slot::<T::EthSpec>(slot);
```

Comment:
Not a reliable way to get the attester slashing fork. One can have an attester slashing electra with an attestation data from deneb. Instead we should implement `fork_name_unchecked` in `slashing`

Reply:
Good catch

> Why aren't we always calculating the fork based on slot_clock.now() here?

I think this makes sense considering return types could be mixed for a given `Eth-Consensus-Version` as pointed out here  https://github.com/sigp/lighthouse/pull/5744#discussion_r1671810990

Reply:
raised a spec issue as well https://github.com/ethereum/beacon-APIs/issues/458